src/Store.js

```diff
@@ -0,0 +1,50 @@
+import invariant from 'invariant';
+import isPlainObject from './utils/isPlainObject';
+
+export default class Store {
+  constructor(reducer, initialState) {
+    invariant(
+      typeof reducer === 'function',
+      'Expected the reducer to be a function.'
+    );
+
+    this.state = initialState;
+    this.listeners = [];
+    this.replaceReducer(reducer);
+  }
+
+  getReducer() {
+    return this.reducer;
+  }
+
+  replaceReducer(nextReducer) {
+    this.reducer = nextReducer;
+    this.dispatch({ type: '@@INIT' });
+  }
+
+  dispatch(action) {
```

Comment:
At this point this method is more like a "performAction" than a "dispatch".
The code reads like "perform an action which modifies state. Then notify listeners that state has changed".